### PR TITLE
feat: devcontainer

### PR DESCRIPTION
**Pull request by mohdnr**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "[terraform]": {
+      "editor.formatOnSave": true
+    }
+  },
+  "features": {
+		"docker-from-docker": {
+			"version": "latest",
+			"moby": true
+		}
+	},
```
Comment (patheard) [inline .devcontainer/devcontainer.json]: nit: formatting

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+RUN curl -Lo terraform.zip https://releases.hashicorp.com/terraform/"${TERRAFORM_VERSION}"/terraform_"${TERRAFORM_VERSION}"_linux_"$(dpkg --print-architecture)".zip \
+    && unzip terraform.zip \
+    && mv terraform /usr/local/bin/ \
+    && rm terraform.zip
+
+# Install Terragrunt
+ARG TERRAGRUNT_SRC=https://github.com/gruntwork-io/terragrunt/releases/download
+RUN curl -Lo terragrunt ${TERRAGRUNT_SRC}/v"${TERRAGRUNT_VERSION}"/terragrunt_linux_"$(dpkg --print-architecture)" \
+    && chmod +x terragrunt \
+    && mv terragrunt /usr/local/bin/
```
Comment (patheard) [inline .devcontainer/Dockerfile]: Could we try replacing these with the [devcontainer `terraform` feature](https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/terraform.md)?

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+RUN pip3 install --upgrade requests setuptools \
+    && pip3 install --upgrade botocore checkov
+
+RUN curl -sL https://awscli.amazonaws.com/awscli-exe-linux-x86_64-${AWS_CLI_VERSION}.zip -o awscliv2.zip \
+    && unzip awscliv2.zip \
+    && aws/install \
+    && rm -rf \
+        awscliv2.zip \
+        aws \
+        /usr/local/aws-cli/v2/*/dist/awscli/examples
```
Comment (patheard) [inline .devcontainer/Dockerfile]: Same idea for the [`awscli` devcontainer feature](https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/awscli.md) if possible.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (mohdnr) [reply .devcontainer/Dockerfile]: amazing. nice find! 🎉

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (mohdnr) [reply .devcontainer/devcontainer.json]: this one always gets me on new repos!

**Review by patheard — APPROVED** [review]

**Pull request merged**